src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -177,6 +187,9 @@ public SwerveDriveKinematics getKinematics() {
         }
 
         public Pose2d getPose() {
+                if(Robot.isSimulation() && !DriverStationSim.getAutonomous()) {
+                        return new Pose2d(new Translation2d(-robotPose.getY(), robotPose.getX()), robotPose.getRotation());
+                }
```

Comment:
I also was thinking that ideally getPose should not be any different in simulation vs normal as all of the differences should just be baked into the pose itself